src/main/java/at/hannibal2/skyhanni/features/chat/Translator.kt

```diff
@@ -0,0 +1,217 @@
+                    )
+                }
+            } finally {
+                client.close()
+            }
+            return JsonObject()
+        }
+
+        private fun getTranslationToEnglish(message: String): String {
+            val url =
+                "https://translate.googleapis.com/translate_a/single?client=gtx&sl=auto&tl=en&dt=t&q=" + URLEncoder.encode(
+                    message,
+                    "UTF-8"
+                )
+
+            var messageToSend = ""
+            val layer1 = getJSONResponse(url).asJsonArray
+            val language = layer1[2].toString()
+            if (language == "en") return "Unable to translate!"
+
+            val layer2 = layer1[0] as JsonArray
+
+            for (layer3 in layer2) {
+                val arrayLayer3 = layer3 as JsonArray
+                val sentence = arrayLayer3[0].toString()
+                val sentenceWithoutQuotes = sentence.substring(1, sentence.length - 1)
+                messageToSend = "$messageToSend$sentenceWithoutQuotes"
+            }
+            messageToSend = "$messageToSend §7(Language: $language)"
+
+            return URLDecoder.decode(messageToSend, "UTF-8").replace("\\", "")
+
+        }
+
+        private fun getTranslationFromEnglish(message: String, lang: String): String {
+            val url =
+                "https://translate.googleapis.com/translate_a/single?client=gtx&sl=en&tl=$lang&dt=t&q=" + URLEncoder.encode(
+                    message,
+                    "UTF-8"
+                )
+
+            val layer1 = getJSONResponse(url).asJsonArray
+            val layer2 = layer1[0] as JsonArray
+
+            val firstSentence = (layer2[0] as JsonArray).get(0).toString()
+            var messageToSend = firstSentence.substring(0, firstSentence.length - 1)
+            for (sentenceIndex in 1..<layer2.size()) {
+                val sentence = (layer2[sentenceIndex] as JsonArray).get(0).toString()
+                val sentenceWithoutQuotes = sentence.substring(1, sentence.length - 1)
+                messageToSend = "$messageToSend$sentenceWithoutQuotes"
+            } // The first translated sentence only has 1 extra char at the end, but sentences after it need 1 at the front and 1 at the end removed in the substring
+            messageToSend = messageToSend.substring(1, messageToSend.length - 1)
+            return URLDecoder.decode(messageToSend, "UTF-8").replace("\\", "")
+        }
+
+        fun toEnglish(args: Array<String>) {
+            if (!SkyHanniMod.feature.chat.translator) return
+            var message = ""
+            for (i in args) {
+                message = "$message$i "
+            }
+
+            val translation = getTranslationToEnglish(message)
+
+            if (translation == "Unable to translate!") LorenzUtils.chat("§c[SkyHanni] Unable to translate message :(")
+            else LorenzUtils.chat("§6[SkyHanni] §eFound translation: $translation")
```

Comment:
pls fix color code

Reply:
which color code needs changing? the §6?

Reply:
correct